pandas/tools/pivot.py

```diff
@@ -436,18 +439,52 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
     data.update(zip(rownames, index))
     data.update(zip(colnames, columns))
 
+    if values is None and aggfunc is not None:
+        raise ValueError("`aggfunc` cannot be used with `values`.")
+
     if values is None:
         df = DataFrame(data)
         df['__dummy__'] = 0
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=len, margins=margins, dropna=dropna)
-        return table.fillna(0).astype(np.int64)
+        table = table.fillna(0).astype(np.int64)
+
     else:
         data['__dummy__'] = values
         df = DataFrame(data)
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=aggfunc, margins=margins, dropna=dropna)
-        return table
+
+    # Post-process
+    table = _normalize(table, normalize=normalize, margins=margins)
+
+    return table
+
+
+def _normalize(table, normalize, margins):
+
+    if isinstance(normalize, str) and \
+            normalize not in ['all', 'index', 'columns']:
```

Comment:
use a dictionary where you define each of these operations as a lamba function.

Then if its not in the dict you can raise the invalid argument.

Then just actually execute it...something like:


```
normalizers = {
   'all' : lambda x: x / x.sum(axis=1).sum(axis=0),
.....
}
normalizers[True] = normalizers['all']

try:
   f = normalizes[normalize]
except KeyError:
    raise ValueError(.....)

table = f(table)
```